Mathlib/GroupTheory/Coxeter/StandardGeometricRepresentation.lean

```diff
@@ -0,0 +1,716 @@
+  _ = (b * (b * (a * b) ^ m)) v'       := by rw [mul_assoc]
+  _ = (b * ((b * a) ^ m * b)) v'       := by
+    congr 2
+    exact (SemiconjBy.eq (SemiconjBy.pow_right (by unfold SemiconjBy; group) m))
+  _ = (b * (b * a) ^ m * b) v'         := by rw [mul_assoc]
+  _ = (b * (b * a) ^ m) (b v')         := LinearMap.mul_apply _ _ _
+  _ = (b * (b * a) ^ m) (-v')          := congrArg _ (M.orthoReflection_apply_self hv')
+  _ = -((b * (b * a) ^ m) v')          := map_neg _ _
+  _ = -(b (((b * a) ^ m) v'))          := congrArg _ (LinearMap.mul_apply _ _ _)
+  _ = -(b v')                          := by
+    congr
+    apply M.orthoReflection_mul_orthoReflection_pow_order_apply_v hm hv' hv
+    · rwa [← M.standardBilinForm_comm v v']
+  _ = -(-v')                           := congrArg _ (M.orthoReflection_apply_self hv')
+  _ = v'                               := neg_neg v'
+
+private lemma can_decomp_into_parallel_and_orthogonal {v v' : V} (w : V) {m : ℕ}
+    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvv' : ⟪v, v'⟫ = -cos (π / m)) (hm : m > 1) :
+    ∃ μ₁ μ₂ : ℝ, ⟪v, w - μ₁ • v - μ₂ • v'⟫ = 0 ∧ ⟪v', w - μ₁ • v - μ₂ • v'⟫ = 0 := by
+  use (1 / (sin (π / m)) ^ 2) * (⟪v, w⟫ + cos (π / m) * ⟪v', w⟫)
+  use (1 / (sin (π / m)) ^ 2) * (⟪v', w⟫ + cos (π / m) * ⟪v, w⟫)
+  -- Expand everything out.
+  simp only [mul_add, LinearMap.map_sub, LinearMap.map_add, LinearMap.map_smul, smul_eq_mul]
+  -- Use known values of bilinear form.
+  rw [(by rw [← M.isSymm_standardBilinForm.eq v' v]; simp : ⟪v', v⟫ = ⟪v, v'⟫)]
+  simp only [hv, hv', hvv']
+  field_simp [sin_pi_div_m_ne_zero hm]
+  ring_nf
+  constructor
+  all_goals {
+    rw [Real.sin_sq]
+    ring
+  }
+
+private lemma fixed_of_orthogonal {v v' : V} (w : V) {m : ℕ}
+    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvw : ⟪v, w⟫ = 0) (hv'w : ⟪v', w⟫ = 0) :
+    (((r hv) * (r hv')) ^ m) w = w := by
+  induction' m with m ih
+  · simp
+  · rw [pow_succ', LinearMap.mul_apply, ih, LinearMap.mul_apply]
+    dsimp [orthoReflection]
+    simp [hvw, hv'w]
+
+private lemma orthoReflection_mul_orthoReflection_pow_order {v v' : V} {m : ℕ}
+    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvv' : ⟪v, v'⟫ = -cos (π / m)) (hm : m ≠ 1) :
+    ((r hv) * (r hv')) ^ m = 1 := by
+  rcases Nat.lt_or_gt_of_ne hm with mlt | mgt
+  · simp [Nat.lt_one_iff.mp mlt]
+  · apply LinearMap.ext
+    intro w
+    rcases M.can_decomp_into_parallel_and_orthogonal w hv hv' hvv' mgt with ⟨μ₁, μ₂, hμ⟩
+    set! w' := w - μ₁ • v - μ₂ • v' with hw'
+    rw [← hw'] at hμ
+    rcases hμ with ⟨h₁, h₂⟩
+    have h₃ : w = w' + μ₁ • v + μ₂ • v' := by rw [hw']; abel
+    simp only [h₃, LinearMap.map_add, LinearMap.map_smul, LinearMap.one_apply]
+    congr
+    · exact M.fixed_of_orthogonal w' hv hv' h₁ h₂
+    · exact M.orthoReflection_mul_orthoReflection_pow_order_apply_v mgt hv hv' hvv'
+    · exact M.orthoReflection_mul_orthoReflection_pow_order_apply_v' mgt hv hv' hvv'
+
+end CoxeterMatrix
```

Comment:
This PR is quite large. Do you think this is a natural point to split it? So that reviews can focus on getting the first half right, and then in a second batch focus on the 2nd half?

I see that there are some long (but documented! thanks!) proofs below. So I expect that people will want to look at them closely.